docs/proposals/improve-status-in-CAPI-resources.md

```diff
@@ -0,0 +1,958 @@
+
+---
+title: Proposal Template
+authors:
+- "@fabriziopandini"
+reviewers:
+- "add"
+creation-date: 2024-07-17
+last-updated: 2024-07-17
+status: provisional
+see-also:
+- ...
+---
+
+# Improving status in CAPI resources
+
+## Table of Contents
+
+- [Improving status in CAPI resources](#improving-status-in-capi-resources)
+  - [Table of Contents](#table-of-contents)
+- [Summary](#summary)
+- [Motivation](#motivation)
+    - [Goals](#goals)
+    - [Non-Goals/Future Work](#non-goalsfuture-work)
+  - [Proposal](#proposal)
+    - [Readiness and Availability](#readiness-and-availability)
+    - [Transition to K8s API conventions aligned conditions](#transition-to-k8s-api-conventions-aligned-conditions)
+    - [Changes to Machine resource](#changes-to-machine-resource)
+      - [Machine Status](#machine-status)
+        - [Machine (New)Conditions](#machine-newconditions)
+      - [Machine Spec](#machine-spec)
+      - [Machine Print columns](#machine-print-columns)
+    - [Changes to MachineSet resource](#changes-to-machineset-resource)
+      - [MachineSet Status](#machineset-status)
+      - [MachineSet (New)Conditions](#machineset-newconditions)
+      - [MachineSet Print columns](#machineset-print-columns)
+    - [Changes to MachineDeployment resource](#changes-to-machinedeployment-resource)
+      - [MachineDeployment Status](#machinedeployment-status)
+      - [MachineDeployment (New)Conditions](#machinedeployment-newconditions)
+      - [MachineDeployment Print columns](#machinedeployment-print-columns)
+    - [Changes to Cluster resource](#changes-to-cluster-resource)
+      - [Cluster Status](#cluster-status)
+        - [Cluster (New)Conditions](#cluster-newconditions)
+      - [Cluster Spec](#cluster-spec)
+      - [Cluster Print columns](#cluster-print-columns)
+    - [Changes to KubeadmControlPlane (KCP) resource](#changes-to-kubeadmcontrolplane-kcp-resource)
+      - [KubeadmControlPlane Status](#kubeadmcontrolplane-status)
+      - [KubeadmControlPlane (New)Conditions](#kubeadmcontrolplane-newconditions)
+      - [KubeadmControlPlane Print columns](#kubeadmcontrolplane-print-columns)
+    - [Changes to MachinePool resource](#changes-to-machinepool-resource)
+    - [Changes to Cluster API contract](#changes-to-cluster-api-contract)
+
+# Summary
+
+This document defines how status across all Cluster API resources is going to evolve in the v1beta2 API version, focusing on
+improving usability and consistency across different resources in CAPI and with the rest of the ecosystem.
+
+# Motivation
```

Comment:
I think one example of this could be that eventually something like AvailableGates should be also configurable via the ClusterClass. So that "end users" don't have to worry about how Availability should be computed for a certain class of Clusters

Reply:
Added a goal for this use case, PTAL
Also please note that there is already a note about the fact that in future we might consider extending CC to allow the definition of AvailabilityGates (first we need to get them in place 😅 )

Reply:
Also added to https://github.com/kubernetes-sigs/cluster-api/issues/11105 for tracking